src/hooks/useVenueAccess.ts

```diff
@@ -22,33 +22,37 @@ export const useVenueAccess = (
     onDenyAccess && onDenyAccess();
   }, [onDenyAccess, venue]);
 
+  const logout = useCallback(() => {
+    firebase
+      .auth()
+      .signOut()
+      .finally(() => {
+        denyAccess();
+      });
+  }, [denyAccess]);
```

Comment:
Is venue access designed to be used only with the main venue of the event, or any venue within the event (my understanding was the latter).

If it is for any venue within an event, then we probably don't want to log the user out of the entire platform, just because they weren't able to enter one venue.

Reply:
Yeah I'm still trying to figure out the correct UX. I'll probably send them to the landing page instead of logging them out.

Reply:
If it’s possible to check + redirect to the parentVenueId if it exists, otherwise the ‘root parent venue’, that would probably be good UX. that way if they are ‘deep’ in nested venues, they will only go back to the one they were trying to enter from or similar.

Reply:
After `denyAccess` is called a modal pops up that informs the user that he has no access after he clicks "Ok" he gets redirected to the landing page where he can type the password.

Reply:
Cool, sounds look a good flow